viam-cartographer.go

```diff
@@ -231,7 +233,7 @@ func (cartoSvc *cartographerService) GetPosition(ctx context.Context) (spatialma
 func (cartoSvc *cartographerService) GetPointCloudMap(ctx context.Context) (func() ([]byte, error), error) {
 	ctx, span := trace.StartSpan(ctx, "viamcartographer::cartographerService::GetPointCloudMap")
 	defer span.End()
-
+	cartoSvc.mapTimestamp = time.Now()
```

Comment:
[major] I believe you are missing some logic regarding when to update the timestamp/call a new map when running in localization mode. Think about the two flows:
```
LOCALIZATION
Startup: timestamp = timestamp1
First call to GetProperties: timestamp = timestamp1
Timestamp has changed: Call GetMap: return map, no update to timestamp
Second call to GetProperties: timestamp = timestamp1
Timestamp has not changed: no call to GetMap
....
Reconfiguration occurs  timestamp = timestamp2
Nth call to GetProperties:timestamp = timestamp2
Timestamp has changed: Call  GetMap: return map, no update to timestamp
N+1th call to GetProperties:timestamp = timestamp2
Timestamp has not changed: no call to GetMap
NON-LOCALIZATION
Startup: timestamp = timestamp1
First call to GetProperties: timestamp = timestamp1
Timestamp has changed: Call GetMap: return map, update timestamp = timestamp2
Second call to GetProperties: timestamp = timestamp2
Timestamp has changed: Call GetMap: return map, update timestamp = timestamp3
....
Reconfiguration occurs  timestamp = timestamp_n
Nth call to GetProperties:timestamp = timestamp_n
Timestamp has changed: Call  GetMap: return map, update timestamp = timestamp_n+1
N+1th call to GetProperties:timestamp = timestamp_n+1
Timestamp has not changed: Call  GetMap: return map, update timestamp = timestamp_n+2
```